### ISSUE TITLE
bug: could not find refactor `nil` when using `select_refactor`

### ISSUE THREAD

**Issue by ViRu-ThE-ViRuS** [ThePrimeagen/refactoring.nvim#269]
```
E5108: Error executing lua ...ack/packer/opt/refactoring.nvim/lua/refactoring/init.lua:20: Could not
 find refactor nil.  You can get a list of all refactors from require("refactoring").get_refactors()

stack traceback:
        [C]: in function 'error'
        ...ack/packer/opt/refactoring.nvim/lua/refactoring/init.lua:20: in function 'refactor'
        ...ack/packer/opt/refactoring.nvim/lua/refactoring/init.lua:48: in function 'select_refactor
'
        [string ":lua"]:1: in main chunk
```

config:
```lua 
...

        use {
            'ThePrimeagen/refactoring.nvim',
            after = 'nvim-treesitter',
            config = function()
                require('refactoring').setup()
                vim.api.nvim_set_keymap(
                    "v",
                    "<leader>rr",
                    ":lua require('refactoring').select_refactor()<CR>",
                    { noremap = true, silent = true, expr = false }
                )
             end }


...
```

I can see its happening in `init.lua:20` as in `select_refactor`, the selection is async, and is triggered after `refactor` is called with a null param.

**Comment by pranavrao145** [comment]
@ViRu-ThE-ViRuS thanks for opening an issue. This has been fixed in #273 - take a look and let us know if it's fixed!

**Comment by ViRu-ThE-ViRuS** [comment]
yup all good now!
